ngi_reports/stockholm/project_summary.py

```diff
@@ -76,94 +80,120 @@ def __init__(self, config, LOG, working_dir, **kwargs):
-            if tmp_method not in seq_method_info:
-                seq_method_info.append(tmp_method)
-
+            if tmp_method not in seq_methods.keys():
+                seq_methods[tmp_method] = alphabets[len(seq_methods.keys())]
+            self.flowcell_info[fc_nm]['seq_meth'] = seq_methods[tmp_method]
+            
+            ## Collect quality info for samples and collect lanes of interest
+            for stat in fc_illumina.get('Demultiplex_Stats',{}).get('Barcode_lane_statistics',[]):
+                try:
+                    sample, lane = (stat['Sample ID'], stat['Lane'])
+                    if stat['Project'].replace('__','.') != self.project_name:
+                        continue
+                    ## to put in a empty dict for the first time
+                    sample_qval[sample] = sample_qval.get(sample,{})
+                    try:
+                        sample_qval[sample]['{}_{}'.format(lane, fc_nm)] = {'qval': float(stat.get('% of >= Q30 Bases (PF)')),
+                                                                        'bases': int(stat.get('# Reads').replace(',',''))*int(run_setup.split('x')[-1])}
+                    except TypeError:
+                        pass
+                    # collect lanes to proceed later
+                    if lane not in self.flowcell_info[fc_name]['lanes']:
+                        self.flowcell_info[fc_name]['lanes'][lane] = {'id': lane}
+                except KeyError:
+                    continue
+            
+            ## proceed through the lane summary and get rquired info
+            for lane in self.flowcell_info[fc_name]['lanes'].keys():
+                # for a miseq run there will be only one lane named 'A'
+                lane_sum = fc_run_summary.get(lane, fc_run_summary.get('A',{}))
+                self.flowcell_info[fc_name]['lanes'][lane]['cluster'] = self.get_lane_info('Clusters PF',lane_sum,run_setup[0],True)
+                self.flowcell_info[fc_name]['lanes'][lane]['phix'] = self.get_lane_info('% Error Rate',lane_sum,run_setup[0])
+                self.flowcell_info[fc_name]['lanes'][lane]['avg_qval'] = self.get_lane_info('% Bases >=Q30',lane_sum,run_setup[0])
+            
         ## give proper section name for the methods
-        seq_method_info = [seq_method_info[c].replace("SECTION",alphabets[c]) for c in range(len(seq_method_info))]
-        self.project_info['sequencing_methods'] = "\n\n".join(seq_method_info)
-
-
-
-
-
-
-
-
-
-
-
-
-
-
-
+        self.project_info['sequencing_methods'] = "\n\n".join([m.replace("SECTION",seq_methods[m]) for m in seq_methods])
+        
+        ## calculate average Q30 over all lanes and flowcell
+        for sample in sample_qval:
+            try:
+                qinfo = sample_qval[sample]
+                self.samples_info[sample]['qscore'] = round(sum([(qinfo[k]['qval']/100)*qinfo[k]['bases'] for k in qinfo])*100/sum([qinfo[k]['bases'] for k in qinfo]), 2)
```

Comment:
Wow, this line made my head hurt. Might be worth splitting it up for readability..?

I _think_ this is equivalent...
```python
total_qvalsbp = 0
total_bases = 0
for k in qinfo:
    total_qvalsbp += qinfo[k]['qval'] * qinfo[k]['bases']
    total_bases += qinfo[k]['bases']
avg_qval = total_qvalsbp /  total_bases
self.samples_info[sample]['qscore'] = round(avg_qval, 2)
```

I think you had a `*100` followed by a `/100` too, which cancelled each other out. Definitely check that the above gives the same result if you do use it :wink: